docs/contracts/v4/guides/15-single-sided-liquidity.mdx

```diff
@@ -0,0 +1,423 @@
+
+Depending on our personal preferences we can either err on the higher or lower side of our target by adding or subtracting the `tickSpacing` if the initializable Tick is lower or higher than the theoretically closest Tick.
+
+### Lower Tick
+
+We now find the lower Tick by subtracting the tickSpacing from the upper Tick:
+
+```solidity
+int24 tickLower = tickUpper - TICK_SPACING;
+```
+
+If the price difference is too low, the lower tick may be left of the current Tick of the Pool.
+In that case we would not be able to provide single side liquidity.
+We can either throw an Error or increase our Position by one Tick.
+
+```solidity
+if (tickLower <= currentTick) {
+    tickLower += TICK_SPACING;
+    tickUpper += TICK_SPACING;
+}
+```
+
+We now have a lower and upper Tick for our Position, next we need to construct and mint it.
+
+## Creating the Single Side Liquidity Position
+
+We will first calculate the liquidity corresponding to our token1, tickLower and tickUpper.
+
+```solidity
+uint128 liquidity = LiquidityAmounts.getLiquidityForAmount1(
+    TickMath.getSqrtPriceAtTick(tickLower),
+    TickMath.getSqrtPriceAtTick(tickUpper),
+    token1Amount
+);
+```
+
+And then, we recalculate actual token1 necessary for that liquidity as we round up the liqudity to the nearest number and that could have changed the required token1.
+
+```solidity
+uint256 requiredToken1 = LiquidityAmounts.getAmount1ForLiquidity(
+    TickMath.getSqrtPriceAtTick(tickLower),
+    TickMath.getSqrtPriceAtTick(tickUpper),
+    liquidity
+);
+```
+
+Finally, we have to set up upper limits for Amount1 and Amount0 accounting for slippage.
+
+```solidity
+uint256 amount1Max = requiredToken1 + 1_000; // 0.001 USDC buffer
+uint256 amount0Max = 1e14; // small ETH buffer
+bytes memory hookData = new bytes(0);
+```
+
+### Minting the Position
+
+We approve token transfers on behalf of msg.sender() by the PositionManager:
+
+```Solidity
+if (!currency1.isAddressZero()) {
+    IERC20(Currency.unwrap(currency1)).approve(address(PERMIT2), type(uint256).max);
+    PERMIT2.approve(Currency.unwrap(currency1), address(posm), type(uint160).max, type(uint48).max);
+}
+```
+
+And now mint a new position using PositionManager's mint function:
+
+```solidity
+IPositionManager(address(posm)).mint(
+    pool, tickLower, tickUpper, liquidity, amount0Max, amount1Max, msg.sender, block.timestamp + 60, hookData
+);
+```
+
+We can find the Contract address on the official [Uniswap GitHub](https://docs.uniswap.org/contracts/v4/deployments).
+For local development, the contract address is the same as the network we are forking from.
+So if we are using a local fork of mainnet like described in the [Local development guide](https://docs.uniswap.org/sdk/v3/guides/local-development), the contract address would be the same as on mainnet.
+
+Here is the full code example for these code snippets:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "forge-std/Script.sol";
```

Comment:
i cant highlihgt this whole section, but yeah for minting a position lets just link to our other guides

Reply:
realistically this guide should simply focus on "calculating the ticks" for a single sided position